menu.js

```diff
@@ -124,3 +124,44 @@ closePopup.addEventListener('click', () => {
   popupContainer.style.display = 'none';
 });
 // End popup window
+
+// Form validation JS
+
+const contactForm = document.getElementById('contact-form');
+const email = document.getElementById('email');
+const errorMessage = document.querySelector('.error-message');
+
+function emailValidation() {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email.value));
+}
+
+function emailLowerCheck() {
+  const arr = email.value.split('');
+  let result = true;
+  arr.forEach((x) => {
+    // eslint-disable-next-line max-len
+    if (!((x.charCodeAt() >= 97 && x.charCodeAt() <= 122) || x.charCodeAt() === 64 || x.charCodeAt() === 46)) {
+      result = false;
+    }
+  });
+  return result;
+}
+
+contactForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+  if (email.value.trim() === '') {
+    errorMessage.style.display = 'block';
+    errorMessage.innerText = 'Email is required!';
+  } else if (emailValidation() === false) {
+    errorMessage.style.display = 'block';
+    errorMessage.innerText = 'This is not a valid email.';
+  } else if (emailLowerCheck() === false) {
+    errorMessage.style.display = 'block';
+    errorMessage.innerText = 'Please write your email in lower case.';
+    errorMessage.scrollIntoView(true);
+  } else {
+    errorMessage.style.display = 'none';
+    contactForm.submit();
+  }
+});
```

Comment:
[OPTIONAL] I think we could wrap all of our JS code in a `main` function. And the call `main` once the web app is finished loading. This will ensure that it works correctly. Imagine that the script loads, but maybe some of the elements on the document aren't ready yet. Then it could fail to select these.

Start the script running once the page has finished loading like this: `window.onload = main`

Reply:
Thanks, @D4nielJ! I added your suggestion.